query/test/unit/test_links_file_transform.py

```diff
@@ -0,0 +1,104 @@
+                        'output_file_uuids': ['file_4', 'file_5'], 'protocol_uuids': ['protocol_0', 'protocol_1']}
+        process = LinksFileTransform().format_process_info(mock_links['links'][0])
+        assert process == mock_process
+
+    def test_get_child_process_uuids_returns_correct_ids(self):
+        with self.db.transaction() as (cursor, tables):
+            child_processes = LinksFileTransform().get_child_process_uuids(tables, ['file_4'])
+        expected_result = ['process_1', 'process_2']
+        self.assertCountEqual(expected_result, child_processes)
+
+    def test_get_parent_process_uuids_returns_correct_ids(self):
+        with self.db.transaction() as (cursor, tables):
+            parent_processes = LinksFileTransform().get_parent_process_uuids(tables, ['file_1', 'file_9'])
+        expected_result = ['process_3', 'process_2']
+        self.assertCountEqual(expected_result, parent_processes)
+
+    @patch.object(Tables, 'process_links')
+    def test_create_processes_calls_insert_correct_number_times(self, mock_tables):
+        mock_process = {'process_uuid': 'process_6', 'input_file_uuids': ['file_11', 'file_12', 'file_13'],
+                        'output_file_uuids': ['file_14', 'file_15'], 'protocol_uuids': ['protocol_10', 'protocol_11']}
+        LinksFileTransform.create_processes(mock_tables, mock_process)
+        assert mock_tables.process_links.insert.call_count == 7
+
+    @patch.object(Tables, 'process_links')
+    def test_link_parent_and_child_processes_called_correct_number_of_times(self, mock_tables):
+        link = {
+            "process": "process_6",
+            "inputs": [
+                "file_11",
+                "file_12",
+            ],
+            "input_type": "biomaterial",
+            "outputs": [
+                "file_14",
+                "file_15"
+            ],
+            "output_type": "file",
+            "protocols": [
+                {
+                    "protocol_type": "library_preparation_protocol",
+                    "protocol_id": "protocol_10"
+                },
+                {
+                    "protocol_type": "sequencing_protocol",
+                    "protocol_id": "protocol_11"
+                }
+            ]
+        }
+        with self.db.transaction() as (cursor, tables):
+            LinksFileTransform().links_file_transformer(tables, [link])
+
+        mock_process = {'process_uuid': 'process_5', 'input_file_uuids': ['file_6', 'file_7', 'file_8'],
+                        'output_file_uuids': ['file_11', 'file_12'], 'protocol_uuids': ['protocol_13', 'protocol_11'],
+                        'parents': ['process_1'], 'children': ['process_6']}
+        LinksFileTransform.link_parent_and_child_processes(mock_tables, mock_process)
+        assert mock_tables.process_links.insert_parent_child_link.call_count == 2
+
+    def test_get_all_parents(self):
```

Comment:
This test is really an `HCAProcesses` class and therefore an `TestHCAProcesses` concern, more than it is a `(Test)LinksTable` concern

Reply:
👍